metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -91,149 +59,78 @@ final class BuildTargets(
   def setTables(newTables: Tables): Unit = {
     tables = Some(newTables)
   }
-
-  def reset(): Unit = {
-    sourceItemsToBuildTarget.values.foreach(_.clear())
-    sourceItemsToBuildTarget.clear()
-    buildTargetInfo.clear()
-    javaTargetInfo.clear()
-    scalaTargetInfo.clear()
-    inverseDependencies.clear()
-    buildTargetSources.clear()
-    buildTargetGeneratedDirs.clear()
-    inverseDependencySources.clear()
-    sourceJarNameToJarFile.clear()
-    isSourceRoot.clear()
-  }
-  def sourceItems: Iterable[AbsolutePath] = sourceItemsToBuildTarget.keys
+  def sourceItems: Iterable[AbsolutePath] =
+    data.iterable.flatMap(_.sourceItemsToBuildTarget.keys)
   def sourceItemsToBuildTargets
       : Iterator[(AbsolutePath, JIterable[BuildTargetIdentifier])] =
-    sourceItemsToBuildTarget.iterator
+    data.fromIterators(_.sourceItemsToBuildTarget.iterator)
+  private def allBuildTargetIdsInternal
+      : Iterator[(TargetData, BuildTargetIdentifier)] =
+    data.fromIterators(d => d.allBuildTargetIds.iterator.map((d, _)))
+  def actualSource(path: AbsolutePath): Option[TargetData.MappedSource] =
```

Comment:
Could we change the name a bit? Not sure `actualSource` means in this case.

Reply:
Done (it's `mappedTo` now).